File: registers/ac_register/ACRegister.py
from m5.params import *
from m5.SimObject import SimObject

class ACRegister(SimObject):
    type = 'ACRegister'
    cxx_header = "CAProject/registers/ac_register/ac_register.hh"
    cxx_class = "gem5::ACRegister"

    ac = Param.UInt16(0, "AC stored in the register")

File: registers/ac_register/ac_register.cc
#include "CAProject/registers/ac_register/ac_register.hh"

namespace gem5
{

ACRegister::ACRegister(const ACRegisterParams &params) : 
    SimObject(params),
    ac(params.ac)
{
}

void
ACRegister::setValue(uint16_t val)
{
    ac = val;
}

void
ACRegister::increment()
{
    ac++;
}

void
ACRegister::clear()
{
    ac = 0;
}

uint16_t
ACRegister::getValue()
{
    return ac;
}

} // namespace gem5
